### PR TITLE
feat(vscode): defined DartFrogApplicationRegistry

### PR DESCRIPTION
**Pull request by alestiago**
## Status

**READY**

## Description

Changes:
- Defines DartFrogApplicationRegistry
- Adds applicationsRegistry member to DartFrogDaemon

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: extensions/vscode/src/daemon/dart-frog-application-registry.ts [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+ * "The Dart VM service is listening on http://127.0.0.1:8181/fQBcSu3OOc8=/"
+ */
+const vmServiceUriMessagePrefix = "The Dart VM service is listening on ";
+
+/**
+ * The prefix of the message that is sent by the Dart Frog Daemon when the
+ * application is starting.
+ *
+ * @example
+ * "Running on \u001b]8;;http://localhost:8080\u001b\\http://localhost:8080\u001b]8;;\u001b\\"
```
Comment (erickzanardo) [inline extensions/vscode/src/daemon/dart-frog-application-registry.ts:29]: Do we need these escaping? They make it so much harder to read 😅

File: extensions/vscode/src/daemon/dart-frog-application-registry.ts [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+  }
+
+  private async retrieveVmServiceUri(requestId: string): Promise<string> {
+    return new Promise<string>((resolve) => {
+      const vmServiceUriEventListener = (event: DaemonEvent) => {
+        if (!isLoggerInfoDaemonEvent(event)) {
+          return;
+        } else if (event.params.requestId !== requestId) {
+          return;
+        }
```
Comment (erickzanardo) [inline extensions/vscode/src/daemon/dart-frog-application-registry.ts:199]:
```suggestion
        if (!isLoggerInfoDaemonEvent(event) || event.params.requestId !== requestId) {
          return;
        }
```

File: extensions/vscode/src/daemon/dart-frog-application-registry.ts [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+        DartFrogDaemonEventEmitterTypes.event,
+        vmServiceUriEventListener.bind(this)
+      );
+    });
+  }
+
+  private async retrieveAddress(requestId: string): Promise<string> {
+    return new Promise<string>((resolve) => {
+      const addressEventListener = (message: DaemonEvent) => {
+        if (!isProgressCompleteDaemonEvent(message)) {
```
Comment (erickzanardo) [inline extensions/vscode/src/daemon/dart-frog-application-registry.ts:225]: Same suggestion

File: extensions/vscode/src/daemon/dart-frog-application-registry.ts (hunk 1 above)
Comment (alestiago) [reply extensions/vscode/src/daemon/dart-frog-application-registry.ts:29]: This is the raw message as it comes from the daemon stdout.

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**